File: tests/rosettacode/ackermann.clj
;; Ackermann function A(3,4)
(defn ackermann [m n]
  (cond
    (zero? m) (inc n)
    (zero? n) (ackermann (dec m) 1)
    :else     (ackermann (dec m) (ackermann m (dec n)))))

(println (str "A(3,4) = " (ackermann 3 4)))

File: tests/rosettacode/ackermann.py
# Ackermann function A(3,4)
import sys
sys.setrecursionlimit(10000)

def ackermann(m, n):
    if m == 0:
        return n + 1
    if n == 0:
        return ackermann(m - 1, 1)
    return ackermann(m - 1, ackermann(m, n - 1))

print(f"A(3,4) = {ackermann(3, 4)}")
